### PR TITLE
Parse decimals in a locale-independent way

### PR DESCRIPTION
**Pull request by AndyZe**
Motivated by these MoveIt and ros2_control issues:

https://github.com/ros-planning/moveit2/issues/1882

https://github.com/ros-controls/ros2_control/pull/921/files

### CODE REVIEW

File: src/model.cpp [hunk 1]
```diff
@@ -35,13 +35,14 @@
...
 #include <algorithm>
+#include <boost/algorithm/string/trim.hpp>
+#include <charconv>
+#include <console_bridge/console.h>
 #include <fstream>
-#include <sstream>
-#include <set>
 #include <limits>
+#include <set>
+#include <sstream>
```
Comment (AndyZe) [inline src/model.cpp:44]: Sorted alphabetically

**Comment by AndyZe** [comment]
I think galactic-testing is failing because this usage of std::to_chars requires C++17

https://stackoverflow.com/a/63964295

**Comment by simonschmeisser** [comment]
You need a pretty recent gcc version for floating point support in `std::to_char`

**Comment by AndyZe** [comment]
Alright, thanks. What do you recommend to do here @simonschmeisser ?

**Comment by simonschmeisser** [comment]
First I'd verify the gcc version as the source of the problem, what does galactic mean in terms of Ubuntu?
Then probably copy paste the helper function from MoveIt core helpers

**Comment by simonschmeisser** [comment]
So it seems a very ugly hack (by actually switching the locale, which is global in c, temporarily) to support floating points in `std::from_char` was added for gcc 11.1.0, Ubuntu 20.04 defaults to 9.4.0. Proper support seems to have been implemented by now but is not actually released yet ( https://github.com/gcc-mirror/gcc/commits/master/libstdc%2B%2B-v3/src/c%2B%2B17/floating_from_chars.cc ).

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ repos:
       - id: clang-format
         name: clang-format
         description: Format files with ClangFormat.
-        entry: clang-format-10
+        entry: clang-format
```
Comment (simonschmeisser) [inline .pre-commit-config.yaml]: I'm not up to date with MoveIt2 development but aren't other repos switching to clang-format-14?

Comment (AndyZe) [reply .pre-commit-config.yaml]: I had to make some kind of change here b/c clang-format-10 is not available on Ubuntu 22, so the first CI run failed. Luckily clang-format-14 is the default version for Ubuntu 22 and it also works for the other ROS distros.

Comment (rhaschke) [reply .pre-commit-config.yaml]: You should explicitly state `clang-format-14` (as in other repos) to ensure that a fixed version is used - even if the default changes.

**Review by rhaschke — APPROVED** [review]

**Pull request merged**